.azure-pipelines/client.yml

```diff
@@ -214,3 +214,27 @@ jobs:
         inputs:
           mergeTestResults: true
           testRunTitle: '$(OSName) on Java $(JavaVersion)'
+
+        # Publish JaCoCo code coverage results from a build
+      - task: PublishCodeCoverageResults@1
+        displayName: 'Publish code coverage report'
+        inputs:
+            codeCoverageTool: 'JaCoCo'
+            summaryFileLocation: ./eng/jacoco-test-coverage/target/site/test-coverage/jacoco.xml
+            reportDirectory: ./eng/jacoco-test-coverage/target/site/test-coverage/
+            failIfCoverageEmpty: true
+
+        # Copy test coverage reports to respective OS and Java version directories
+      - task: CopyFiles@2
+        displayName: 'Copy test coverage reports to OS-JDKversion directory'
+        inputs:
+            sourceFolder: eng/jacoco-test-coverage/target/site/
+            targetFolder: test-coverage-$(OSName)-$(JavaVersion)
+
+        # Publish pipeline artifacts for respective OS and Java versions
+      - task: PublishPipelineArtifact@0
+        displayName: 'Publish test coverage artifact'
+        inputs:
+          artifactName: 'test-coverage-$(OSName)-$(JavaVersion)'
```

Comment:
The `artifactName` still needs to be a short, constant value, such as `coverage` or `test-coverage`